site/src/pages/TaskPage/TaskPage.tsx

```diff
@@ -174,6 +168,84 @@ const TaskPage = () => {
 
 export default TaskPage;
 
+type WorkspaceNotRunningProps = {
+	task: Task;
+};
+
+const WorkspaceNotRunning: FC<WorkspaceNotRunningProps> = ({ task }) => {
+	const queryClient = useQueryClient();
+
+	const { data: parameters } = useQuery({
+		queryKey: ["workspace", task.workspace.id, "parameters"],
+		queryFn: () => API.getWorkspaceParameters(task.workspace),
+	});
+
+	const [workspaceErrorDialog, setWorkspaceErrorDialog] = useState<{
+		open: boolean;
+		error?: ApiError;
+	}>({ open: false });
+
+	const handleError = (error: unknown) => {
+		if (isApiError(error)) {
+			setWorkspaceErrorDialog({
+				open: true,
+				error: error,
+			});
+		} else {
+			displayError(getErrorMessage(error, "Failed to build workspace."));
+		}
+	};
+
+	const mutateStartWorkspace = useMutation({
+		...startWorkspace(task?.workspace, queryClient),
+		onError: (error: unknown) => {
+			handleError(error);
```

Comment:
why the indirection? you can just put the contents of `handleError` here since it's not called anywhere else.